### PR TITLE
SP-317 - Replace Timestamp with Counter

### PR DESCRIPTION
**Pull request by AliAkiltayev**
Replaced timestamp with counter in according with JS library

### CODE REVIEW

**Review by mattgresham — APPROVED** [review]
Looks good to me.

Just remember that to test this with the VAA, the protocol needs to be bumped like in this ticket - https://mx51.atlassian.net/browse/SP-712

File: client/src/main/java/io/mx51/spi/model/Message.java [hunk 1]
```diff
@@ -193,21 +197,18 @@ public boolean getDataBooleanValue(String attribute, boolean defaultIfNotFound)
...
-        final long now = System.currentTimeMillis();
-        final long adjustedTime = now + stamp.getServerTimeDelta();
-        dateTimeStamp = new SimpleDateFormat(DATE_TIME_FORMAT, Locale.US).format(new Date(adjustedTime));
+        stamp.setPosCounter(stamp.getPosCounter()+1);
+        posCounter = stamp.getPosCounter();
+        connId = stamp.getConnId();
+
+        dateTimeStamp = new SimpleDateFormat(DATE_TIME_FORMAT, Locale.US).format(new Date());
 
+        System.out.println(dateTimeStamp);
```
Comment (mattgresham) [inline client/src/main/java/io/mx51/spi/model/Message.java:211]: A small thing, could you remove this output please

**Review by RaymondK-mx51 — CHANGES_REQUESTED** [review]

File: client/src/main/java/io/mx51/spi/model/MessageStamp.java [hunk 2]
```diff
@@ -10,12 +10,13 @@
 
     private Secrets secrets;
 
-    private long serverTimeDelta;
+    private String connId;
 
-    public MessageStamp(String posId, Secrets secrets, long serverTimeDelta) {
+    private Integer posCounter;
```
Comment (RaymondK-mx51) [inline client/src/main/java/io/mx51/spi/model/MessageStamp.java:15]: should this be `int`?

File: client/src/main/java/io/mx51/spi/model/Message.java [hunk 3]
```diff
@@ -31,6 +31,10 @@
     private Map<String, Object> data;
     @SerializedName("datetime")
     private String dateTimeStamp;
+    @SerializedName("pos_counter")
+    private Integer posCounter;
```
Comment (RaymondK-mx51) [inline client/src/main/java/io/mx51/spi/model/Message.java:35]: should this be `int`

**Review by RaymondK-mx51 — COMMENTED** [review]
Should all occurences of `Integer` be changed to `int`?

**Comment by AliAkiltayev** [comment]
> Should all occurrences of `Integer` be changed to `int`?

Ok, got it

**Comment by AliAkiltayev** [comment]
> Should all occurrences of `Integer` be changed to `int`?

How about responseCode in DeviceAddressStatus class? Should I also change it?

File: client/src/main/java/io/mx51/spi/model/Message.java (hunk 3 above)
Comment (RaymondK-mx51) [reply client/src/main/java/io/mx51/spi/model/Message.java]: > > Should all occurrences of `Integer` be changed to `int`?
> 
> How about responseCode in DeviceAddressStatus class? Should I also change it?

Yes Please

**Review by RaymondK-mx51 — APPROVED** [review]

**Pull request merged**